### ISSUE TITLE
requesting new address parameter in license class

### ISSUE THREAD

**Issue by JeffGiroux** [F5Networks/f5-declarative-onboarding#144]
### Is your feature request related to a problem? Please describe.
Use case, BIG-IP in cloud has private address. BIG-IQ sits somewhere else. BIG-IP reaches BIG-IQ via public internet. But...when DO is sent from BIG-IP, it uses private address. When BIG-IQ sees request and "discovered address", it is internal and hence not reachable.

### Describe the solution you'd like
Can DO send license request to BIG-IQ with "address" parameter configurable in "License Class"? Currently "address" is not a configurable parameter. Is this on roadmap? I did not see an existing DO issue. The BIG-IQ API already accepts “address” in the license request API call. But…DO does not have “address”.
 
Reference BIG-IQ API:
https://clouddocs.f5.com/products/big-iq/mgmt-api/v7.1.0/ApiReferences/bigiq_public_api_ref/r_license_assign_revoke.html
“address” = string: The IP address used to add the BIG-IP to BIG-IQ. This is sometimes called the discovery address.
 
Now…look at the DO API reference:
 
Reference DO "License Class" (does not have 'address' field):
https://clouddocs.f5.com/products/extensions/f5-declarative-onboarding/latest/big-iq-licensing.html?highlight=bigipusername#license-class

### Describe alternatives you've considered
The alternative option is to not use DO. Instead, send a regular API call to the BIG-IQ API endpoint using the license/assign revoke. This method allows 'address' to be supplied in payload. However, this is not fit into current orchestration methods of using DO, AS3, TS, etc.

### Additional context
Example declare with new “address” field like this…
```
"myLicense": {
     "class": "License",
     "licenseType": "licensePool",
     "bigIqHost": "10.0.1.200",
     "bigIqUsername": "admin",
     "bigIqPassword": "myPassword1",
     "licensePool": "myPool",
     "skuKeyword1": "key1",
     "skuKeyword2": "key2",
     "unitOfMeasure": "hourly",
     "reachable": true,
     "bigIpUsername": "admin",
     "bigIpPassword": "barbar",
     "address": "200.200.100.100"
},
```

When BIG-IQ receives the license request payload, it will have a discoverable address from the BIG-IP of 200.200.100.100.

**Comment by dstokesf5** [comment]
Have you tried setting `reachable` to false for this use case? https://clouddocs.f5.com/products/extensions/f5-declarative-onboarding/latest/bigiq-examples.html#revoking-and-relicensing-a-big-ip-no-route-from-big-iq

**Comment by JeffGiroux** [comment]
reachable=false does not satisfy the use case. Setting unreachable to false merely tells BIG-IQ that the BIG-IP is unreachable so therefore don't bother trying to reach back out to the BIG-IP discoverable address.

Adding @itlinux for more use case info...

What I need is the ability for DO license class to have the discoverable BIG-IP address be editable. Then, the customer can probably grab the public IP of the BIG-IP (through upstream NAT or cloud elastic IP) and put that as the discoverable BIG-IP address. When BIG-IQ gets the request, it has a reachable public IP from the BIG-IP.

**Comment by dstokesf5** [comment]
I see. The discovered address is a public address. From your description I thought the BIG-IP was still unreachable in this case:
>When BIG-IQ sees request and "discovered address", it is internal and hence not reachable.

**Comment by itlinux** [comment]
That is correct, the BIG-IQ is on Azure diff RG, and the BIG-IP is on Azure diff RG, when the BIG-IP requests the license, and gets granted, the BIG-IQ sees the internal mgmt IP address (not the public IP) therefore, when you click on the IP in the BIG-IQ it does not know how to get back to the BIG-IP, RFC 1819. Hopefully, this helps a bit. Let me know we can connect off-line.

**Comment by dstokesf5** [comment]
Thanks for the additional information. I understand the use case now, and I have created AUTOTOOL-1685 in our internal product backlog for it.
